packages/block-editor/README.md

```diff
@@ -766,6 +766,28 @@ _Parameters_
 -   _props_ `Object`: Optional. Props to pass to the element. Must contain the ref if one is defined.
 -   _options_ `Object`: Optional. Inner blocks options.
 
+### useLayoutClasses
+
+Generates the utility classnames for the given blocks layout attributes.
+This method was primarily added to reintroduce classnames that were removed
+in the 5.9 release (<https://github.com/WordPress/gutenberg/issues/38719>), rather
+than providing an extensive list of all possible layout classes. The plan is to
+have the style engine generate a more extensive list of utility classnames which
+will then replace this method.
```

Comment:
Thanks for thinking this through! I'll update the description to remove references to hypothetical future style engine work 😅 

> Would it be worth coming up with a signature for both of these hooks that is pretty much the same, to create some consistency there?

We wouldn't need the selector for `useLayoutClasses` though, would we? Other than that I'm happy to change it to take the block object.

Reply:
> We wouldn't need the selector for useLayoutClasses though, would we?

Ah, good point. Yeah, I can't think of a circumstance where we'd need the selector for getting the classnames. We might have use for it eventually for where we _output_ the classnames, but not for retrieving them in the first place?

Reply:
Yeah I don't think we'd ever need the selector in `useLayoutClasses`.

I'm also struggling to think of a use for the whole block in this function; we only really need the layout object. We could potentially change it to get the layout definitions from `useSetting( 'layout' )` instead of passing them as a parameter. Would that work everywhere?

In `useLayoutStyles` we need both `layout` and `styles` from attributes, as well as the block name to pass to the layout type's `useLayoutStyles` function, so it can work out whether the block skips serialisation or not.

It's not looking like we can make the signatures very consistent here 😅 

Regarding the doc comment for `useLayoutClasses`, could we simply shorten it to "Generates the utility classnames for the given blocks layout attributes.", or is there any additional info that might be useful here?

Also, I'm thinking we should make these experimental given Layout itself has that status.

Reply:
> we only really need the layout object

Right now that's definitely the case. My only thought was if we wind up eventually needing to use parts of the `spacing` object for potential future classnames (e.g. if we needed to add a `has-gap` classname further down the track, which I think was discussed a little in the spacing presets discussions), then it'd be easier if we already had the object there.

> We could potentially change it to get the layout definitions from useSetting( 'layout' ) instead of passing them as a parameter. Would that work everywhere?

That very well could. I think passing it around is probably a result of some of these functions originally being pure functions instead of React hooks? Now that they're hooks, it's probably better to keep the API signature simpler instead of passing stuff around 🤔

> "Generates the utility classnames for the given blocks layout attributes."

Sounds good to me!

> Also, I'm thinking we should make these experimental given Layout itself has that status.

Good idea, there's still plenty that we're moving around, so would be good to maintain that flexibility 👍